Split docker actions based on visualization target

Now that Superset, Evidence, and Rill can be visualization options, I want to split the makefile to run the docker container based on which tool is handling the viz.

Superset works now with:
```
make docker-build
make docker-run-pipeline
make docker-run-superset
```

I want to continue the same pattern with Rill & Evidence, but I am stuck on how to get it to work. Debugging docker is hell!
```
make docker-build
make docker-run-pipeline
make docker-run-<viz_tool>
```

@gregwdata I am getting the following error when running ```make docker-run-evidence```. Do you mind taking a look at this?

```
cd analyze && mkdir -p data_catalog
cp -r /workspaces/nba-monte-carlo/data/data_catalog/* /workspaces/nba-monte-carlo/analyze/data_catalog
cp: cannot stat '/workspaces/nba-monte-carlo/data/data_catalog/*': No such file or directory
make[1]: Leaving directory '/workspaces/nba-monte-carlo'
make[1]: *** [Makefile:43: evidence-build] Error 1
make: *** [Makefile:50: evidence-visuals] Error 2
make: *** [Makefile:54: docker-run-evidence] Error 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,7 @@
-FROM python:3.9
+FROM nikolaik/python-nodejs:python3.9-nodejs19
+
+ARG SSL_KEYSTORE_PASSWORD
+USER root
 
 WORKDIR /workspaces/nba-monte-carlo
 
@@ -24,9 +27,9 @@ RUN meltano --log-level=debug --environment=docker install extractors
 RUN meltano --log-level=debug --environment=docker install loaders
 RUN meltano --log-level=debug --environment=docker install mappers
 RUN meltano --log-level=debug --environment=docker install utility dbt-duckdb
-RUN meltano --log-level=debug --environment=docker install utility superset
 
 COPY data ./data
 COPY transform ./transform
 COPY visuals ./visuals
 COPY Makefile .
+COPY analyze ./analyze

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM nikolaik/python-nodejs:python3.9-nodejs19
+FROM nikolaik/python-nodejs:python3.9-nodejs19-bullseye
 
 ARG SSL_KEYSTORE_PASSWORD
 USER root
@@ -17,6 +17,9 @@ RUN apt-get update && apt-get install -y \
   python3-venv \
   unzip \
   wget \
+  gfortran \
+  musl-dev \
+  glibc-source \
   && rm -rf /var/lib/apt/lists/* \
   && wget https://github.com/duckdb/duckdb/releases/download/v0.6.0/duckdb_cli-linux-amd64.zip && unzip duckdb_cli-linux-amd64.zip \
   && pip install --no-cache-dir meltano==2.10.0

--- a/Makefile
+++ b/Makefile
@@ -10,24 +10,32 @@ pipeline:
 	meltano invoke dbt-duckdb run-operation elo_rollforward
 	meltano invoke dbt-duckdb build
 
+docker-build:
+	docker build -t mdsbox .
+
+docker-run-pipeline:
+	docker run \
+	 	--env MELTANO_CLI_LOG_LEVEL=WARNING \
+		--env MDS_SCENARIOS=10000 \
+		--env MDS_INCLUDE_ACTUALS=true \
+		--env MDS_LATEST_RATINGS=true \
+		--env MDS_ENABLE_EXPORT=true \
+		--env ENVIRONMENT=docker \
+		mdsbox make pipeline 
+
 superset-visuals:
+	meltano install utility superset
 	meltano invoke superset fab create-admin --username admin --firstname lebron --lastname james --email admin@admin.org --password password
 	meltano invoke superset import-datasources -p visuals/datasources.yml
 	meltano invoke superset import-dashboards -p visuals/dashboards.json
 	meltano invoke superset:ui
 
-docker-build:
-	docker build -t mdsbox .
-
-docker-run:
+docker-run-superset:
 	docker run \
 		--publish 8088:8088 \
 	 	--env MELTANO_CLI_LOG_LEVEL=WARNING \
-		--env MDS_SCENARIOS=1000 \
-		--env MDS_INCLUDE_ACTUALS=true \
-		--env MDS_LATEST_RATINGS=true \
-		--env MDS_ENABLE_EXPORT=true \
-		mdsbox make pipeline superset-visuals 
+		--env ENVIRONMENT=docker \
+		mdsbox make superset-visuals
 
 evidence-build:
 	cd analyze && npm update
@@ -37,3 +45,14 @@ evidence-build:
 
 evidence-run:
 	cd analyze && npm run dev
+
+evidence-visuals:
+	make evidence-build
+	make evidence-run
+
+docker-run-evidence:
+	docker run \
+		--publish 3000:3000 \
+	 	--env MELTANO_CLI_LOG_LEVEL=WARNING \
+		--env ENVIRONMENT=docker \
+		mdsbox make evidence-visuals

--- a/Makefile
+++ b/Makefile
@@ -44,7 +44,7 @@ evidence-build:
 	cp /workspaces/nba-monte-carlo/analyze/data_catalog/mdsbox.db /workspaces/nba-monte-carlo/analyze/
 
 evidence-run:
-	cd analyze && npm run dev
+	cd analyze && npm run dev -- --host 0.0.0.0
 
 evidence-visuals:
 	make evidence-build
@@ -56,3 +56,14 @@ docker-run-evidence:
 	 	--env MELTANO_CLI_LOG_LEVEL=WARNING \
 		--env ENVIRONMENT=docker \
 		mdsbox make evidence-visuals
+
+docker-evidence-test:
+		docker run \
+		--publish 3000:3000 \
+	 	--env MELTANO_CLI_LOG_LEVEL=WARNING \
+		--env MDS_SCENARIOS=10000 \
+		--env MDS_INCLUDE_ACTUALS=true \
+		--env MDS_LATEST_RATINGS=true \
+		--env MDS_ENABLE_EXPORT=true \
+		--env ENVIRONMENT=docker \
+		mdsbox make pipeline evidence-visuals

--- a/meltano.yml
+++ b/meltano.yml
@@ -171,6 +171,11 @@ plugins:
           executable: dbt_extension
       config:
         path: $MELTANO_PROJECT_ROOT/data/data_catalog/mdsbox.db
+    - name: superset
+      variant: apache
+      pip_url: apache-superset>=1.5.0 markupsafe==2.0.1 Werkzeug==2.0.3 WTForms==2.3.0 duckdb-engine==0.6.4
+      config:
+        ENABLE_PROXY_FIX: true
   mappers:
    - name: meltano-map-transformer
      variant: meltano

